src/roomHandlers/fishingRoomHandler.ts

```diff
@@ -0,0 +1,128 @@
+import { RoomHandler } from './roomHandler';
+
+import {
+  Command,
+  CommandHandlersByCommandByPlayerStates,
+  DEFAULT_COMMAND,
+  NarrowedRoomState,
+  NullableCommandHandler,
+  StateDescriptionAccessor,
+} from '../utils/types';
+
+const roomType = 'fishingRoom';
+type TRoomType = typeof roomType;
+type TRoomState = NarrowedRoomState<TRoomType>;
+type TStateDescriptionAccessor = StateDescriptionAccessor<TRoomState>;
+type TCommandHandlersByCommandByPlayerStates =
+  CommandHandlersByCommandByPlayerStates<TRoomType>;
+type TNullableCommandHandler = NullableCommandHandler<TRoomType>;
+
+const stateDescriptionAccessor: TStateDescriptionAccessor = {
+  AtEntrance: ({ fishCaught, brokenRod }) => {
+    const poleText = brokenRod === true ? 'broken' : 'working';
+
+    return {
+      playerStateDescription: `You are on a dock with a ${poleText} fishing pole. Total number of fish caught is ${fishCaught} so far!`,
+      availableCommands: ['fish', 'leave'],
+    };
+  },
+  Fishing: {
+    playerStateDescription: 'Hopefully the fish are hungry today.',
+    availableCommands: ['stop', 'fish'],
+  },
+  Caught: {
+    playerStateDescription: 'Congratulations!!! You caught a fish!!!',
+    availableCommands: ['goToEntrance'],
+  },
+  BrokenLine: {
+    playerStateDescription:
+      'You have a broken fishing line and the fish got away.',
+    availableCommands: ['stop', 'fix'],
+  },
+};
+
+const commandHandlersByCommandByPlayerState: TCommandHandlersByCommandByPlayerStates =
+  {
+    AtEntrance: {
+      fish: (roomState) => ({
+        commandDescription: 'You start to fish',
```

Comment:
Should this statement be conditional? Either `You approach the broken fishing line` or `You pick up the line and start to fish`

Reply:
Updated to be conditional.